editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -3654,6 +3655,12 @@ void TileMapLayerEditor::_node_change(Node *p_node) {
 
 void TileMapLayerEditor::_notification(int p_what) {
 	switch (p_what) {
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (is_visible()) {
+				CanvasItemEditor::update_tool();
```

Comment:
I'm still unsure about this change, as selecting a node changing the tool might be a bit annoying to users.
But I guess we can give it a try and see if anyone complains.